search-service/src/test/kotlin/com/egm/stellio/search/service/AttributeInstanceServiceTests.kt

```diff
@@ -659,6 +662,44 @@ class AttributeInstanceServiceTests : WithTimescaleContainer, WithKafkaContainer
         }
     }
 
+    @Test
+    fun `it should modify attribute instance`() = runTest {
+        val attributeInstance = gimmeAttributeInstance(incomingTemporalEntityAttribute.id)
+        attributeInstanceService.create(attributeInstance)
+
+        val instanceTemporalFragment =
+            loadSampleData("fragments/temporal_instance_fragment.jsonld")
+        val attributeInstancePayload = mapOf(INCOMING_COMPACT_PROPERTY to instanceTemporalFragment.deserializeAsList())
+        val jsonLdAttribute = JsonLdUtils.expandJsonLdFragment(
+            attributeInstancePayload,
+            listOf(APIC_COMPOUND_CONTEXT)
+        ) as ExpandedAttributesInstances
+
+        val temporalEntitiesQuery = gimmeTemporalEntitiesQuery(
+            TemporalQuery(
+                timerel = TemporalQuery.Timerel.AFTER,
+                timeAt = ZonedDateTime.parse("1970-01-01T00:00:00Z")
+            )
+        )
+
+        attributeInstanceService.modifyAttributeInstance(
+            entityId,
+            INCOMING_PROPERTY,
+            attributeInstance.instanceId,
+            jsonLdAttribute
+        ).shouldSucceedAndResult()
+
+        attributeInstanceService.search(temporalEntitiesQuery, incomingTemporalEntityAttribute)
+            .shouldSucceedWith {
+                (it as List<FullAttributeInstanceResult>).single { result ->
+                    result.time == ZonedDateTime.parse("2023-03-13T12:33:06Z") &&
+                        result.payload.deserializeAsMap().any {
+                            it.key == NGSILD_MODIFIED_AT_PROPERTY || it.key == NGSILD_INSTANCE_ID_PROPERTY
+                        }
```

Comment:
very strange check...

Reply:
It's to check that modifiedAt and instanceId are in payload

Reply:
But I'm right it's confused